src/plugins/sails.js

```diff
@@ -54,12 +54,21 @@ io.socket.on('game', function (evData) {
           store.commit('updateReady', evData.data.pNum);
           break;
         case 'Initialize': {
+          const currentRoute = router.currentRoute.value;
+          const isSpectating = currentRoute.name === ROUTE_NAME_SPECTATE;
+
+          // Update state
           store.commit('resetState');
           store.dispatch('updateGameThenResetPNumIfNull', evData.data.game);
-          const gameRoute = `/game/${store.state.game.id}`;
-          const currentRoute = router.currentRoute.fullPath;
-          if (gameRoute !== currentRoute) {
-            router.push(gameRoute);
+          if (isSpectating) {
+            store.commit('setMyPNum', 0); // always spectate as p0
```

Comment:
Is there a follow up to look in to this? Or are we just using this as the baseline and changing based on player vs. observer to show cards, etc?

Reply:
Like whether you spectate as p0 or p1? Look into what?

Reply:
Meaning should there be a special player num for observers or does p0 do what you require? Feels kind of weird to share a pnum despite knowing it’s just a way to get the events/game moves.

Reply:
Okay yeah I hear you. The main reason for using a 'normal' 'valid' pNum is that it leaves all the existing getter logic around indentifying which stuff goes at the bottom of the screen i.e. the player's cards vs the top i.e. the opponent's cards. Making the spectator one or the other player (atm it's just always p0) means we don't need to make all those other getters and such pay attention to whether your pNum is something else that _isn't_ an index in the array of players, which feels simpler to me.

Instead we keep spectating as a getter based on the current route, which lets consumers who care about that difference use it however they need to while letting most of the presentational logic do it's thing in the usual way. Otherwise I think there will be a lot of places that need to handle the pNum being something other than 1 or 0

Reply:
This makes a lot more sense now, thanks. It might make sense to change this long run for clarity but for now, for the MVP, I agree this is the best move for now.